File: solapi-kotlin-example-java/src/main/java/com/solapi/example/SendMmsExample.java
package com.solapi.example;

import com.solapi.sdk.SolapiClient;
import com.solapi.sdk.message.dto.response.MultipleDetailMessageSentResponse;
import com.solapi.sdk.message.model.Message;
import com.solapi.sdk.message.model.MessageType;
import com.solapi.sdk.message.model.StorageType;
import com.solapi.sdk.message.service.DefaultMessageService;

import java.io.File;
import java.io.InputStream;
import java.net.URL;
import java.nio.file.Files;
import java.nio.file.StandardCopyOption;

/**
 * MMS 이미지 첨부 발송 예제
 *
 * 환경변수 설정:
 * - SOLAPI_API_KEY: SOLAPI API 키
 * - SOLAPI_API_SECRET: SOLAPI API 시크릿
 * - SOLAPI_SENDER: 등록된 발신번호
 * - SOLAPI_RECIPIENT: 수신번호
 *
 * 참고: MMS 이미지 규격
 * - 지원 포맷: JPG, JPEG
 * - 최대 용량: 200KB
 * - 권장 해상도: 1000x1000 이하
 */
public class SendMmsExample {

    public static void main(String[] args) {
        // 환경변수에서 설정 로드
        String apiKey = System.getenv("SOLAPI_API_KEY");
        String apiSecret = System.getenv("SOLAPI_API_SECRET");
        String sender = System.getenv("SOLAPI_SENDER");
        String recipient = System.getenv("SOLAPI_RECIPIENT");

        if (apiKey == null || apiSecret == null) {
            System.err.println("SOLAPI_API_KEY and SOLAPI_API_SECRET must be set");
            System.exit(1);
        }
        if (sender == null || recipient == null) {
            System.err.println("SOLAPI_SENDER and SOLAPI_RECIPIENT must be set");
            System.exit(1);
        }

        // SDK 클라이언트 생성
        DefaultMessageService messageService = SolapiClient.INSTANCE.createInstance(apiKey, apiSecret);

        try {
            // 이미지 파일 로드 (리소스에서)
            URL imageUrl = SendMmsExample.class.getClassLoader().getResource("images/sample.jpg");
            if (imageUrl == null) {
                System.err.println("Sample image not found in resources/images/sample.jpg");
                System.err.println("Please add a JPG image file (max 200KB) to run this example.");
                System.exit(1);
            }

            // 임시 파일로 복사 (URL에서 File로 변환)
            File tempFile = File.createTempFile("mms-image", ".jpg");
            tempFile.deleteOnExit();
            try (InputStream is = imageUrl.openStream()) {
                Files.copy(is, tempFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
            }

            // 이미지 업로드
            System.out.println("이미지 업로드 중...");
            String imageId = messageService.uploadFile(tempFile, StorageType.MMS, null);
            System.out.println("이미지 업로드 완료 - imageId: " + imageId);

            // MMS 메시지 생성
            Message message = new Message();
            message.setType(MessageType.MMS);
            message.setFrom(sender);
            message.setTo(recipient);
            message.setText("안녕하세요. MMS 이미지 첨부 메시지입니다.");
            message.setSubject("MMS 제목");
            message.setImageId(imageId);

            // 메시지 발송
            MultipleDetailMessageSentResponse response = messageService.send(message, null);

            System.out.println("MMS 발송 성공!");
            System.out.println("Group ID: " + response.getGroupInfo().getGroupId());

        } catch (Exception e) {
            System.err.println("MMS 발송 실패: " + e.getMessage());
            e.printStackTrace();
        }
    }
}

File: solapi-kotlin-example-kotlin/src/main/kotlin/com/solapi/example/SendMmsExample.kt
package com.solapi.example

import com.solapi.sdk.SolapiClient
import com.solapi.sdk.message.model.Message
import com.solapi.sdk.message.model.MessageType
import com.solapi.sdk.message.model.StorageType
import java.io.File
import java.nio.file.Files
import java.nio.file.StandardCopyOption

/**
 * MMS 이미지 첨부 발송 예제
 *
 * 환경변수 설정:
 * - SOLAPI_API_KEY: SOLAPI API 키
 * - SOLAPI_API_SECRET: SOLAPI API 시크릿
 * - SOLAPI_SENDER: 등록된 발신번호
 * - SOLAPI_RECIPIENT: 수신번호
 *
 * 참고: MMS 이미지 규격
 * - 지원 포맷: JPG, JPEG
 * - 최대 용량: 200KB
 * - 권장 해상도: 1000x1000 이하
 */
fun main() {
    // 환경변수에서 설정 로드
    val apiKey = System.getenv("SOLAPI_API_KEY")
        ?: error("SOLAPI_API_KEY must be set")
    val apiSecret = System.getenv("SOLAPI_API_SECRET")
        ?: error("SOLAPI_API_SECRET must be set")
    val sender = System.getenv("SOLAPI_SENDER")
        ?: error("SOLAPI_SENDER must be set")
    val recipient = System.getenv("SOLAPI_RECIPIENT")
        ?: error("SOLAPI_RECIPIENT must be set")

    // SDK 클라이언트 생성
    val messageService = SolapiClient.createInstance(apiKey, apiSecret)

    try {
        // 이미지 파일 로드 (리소스에서)
        val imageUrl = object {}.javaClass.classLoader.getResource("images/sample.jpg")
            ?: error("Sample image not found in resources/images/sample.jpg. Please add a JPG image file (max 200KB).")

        // 임시 파일로 복사
        val tempFile = File.createTempFile("mms-image", ".jpg").apply {
            deleteOnExit()
        }
        imageUrl.openStream().use { input ->
            Files.copy(input, tempFile.toPath(), StandardCopyOption.REPLACE_EXISTING)
        }

        // 이미지 업로드
        println("이미지 업로드 중...")
        val imageId = messageService.uploadFile(tempFile, StorageType.MMS)
        println("이미지 업로드 완료 - imageId: $imageId")

        // MMS 메시지 생성
        val message = Message(
            type = MessageType.MMS,
            from = sender,
            to = recipient,
            text = "안녕하세요. MMS 이미지 첨부 메시지입니다.",
            subject = "MMS 제목",
            imageId = imageId
        )

        // 메시지 발송
        val response = messageService.send(message)

        println("MMS 발송 성공!")
        println("Group ID: ${response.groupInfo?.groupId}")

    } catch (e: Exception) {
        System.err.println("MMS 발송 실패: ${e.message}")
        e.printStackTrace()
    }
}
